Cql/Cql.CqlToElm/ModelProvider.cs

```diff
@@ -176,8 +176,16 @@ public static bool TryGetTypeInfoFor(this ModelInfo model, string name, out Type
         internal static bool TryFindTypeInfoByName(this IModelProvider provider, string uri,
             string typeName, out TypeInfo? typeInfo, out ModelInfo? model)
         {
-            return provider.TryGetModelFromUri(uri, out model) &
-                    model!.TryGetTypeInfoFor(typeName, out typeInfo);
+            if (provider.TryGetModelFromUri(uri, out model)
```

Comment:
A try function that returns null if it is succesful sounds like a bad design on which we will trip later on. Let me take a look what's going on...

Reply:
No, it cannot happen, maybe you just wanted the nullability warning to go away. We can do that too, by adding [NotNullWhen(true)] to the definition of the TryGetModelFromUri function.